service/src/main/java/io/camunda/service/license/CamundaLicense.java

```diff
@@ -38,8 +40,13 @@ public synchronized String getLicenseType() {
 
   public synchronized void initializeWithLicense(final String license) {
     if (!isInitialized) {
-      isValid = determineLicenseValidity(license);
-      licenseType = getLicenseTypeFromProperty(license);
+      if (StringUtils.isNotBlank(license)) {
```

Comment:
🔧 I'm not a big fan of `StringUtils.isNotBlank(license)`, I'd suggest to use a native approach: `license != null && !license.isBlank()`. To me it's a bit more clear, and we can get rid off the import 😄 

🔧 We can also refactor a bit this method removing some if branching

```
public synchronized void initializeWithLicense(final String license) {
    if (isInitialized) {
        return;
    }

    if (license != null && !license.isBlank()) {
        isValid = determineLicenseValidity(license);
        licenseType = getLicenseTypeFromProperty(license);
    } else {
        LOGGER.error("No license detected when one is expected. Please provide a license.");
    }

    isInitialized = true;
}
```